service-tests/pypi.js

```diff
@@ -0,0 +1,222 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+const {
+  isCommaSeperatedPythonVersions,
+  isSemver,
+} = require('./helpers/validators');
+
+const t = new ServiceTester({ id: 'pypi', title: 'PyPi badges' });
+module.exports = t;
+
+
+/*
+  Note:
+  Download statistics are no longer available from pypi
+  it is exptected that the download badges all show
+  'no longer available'
+*/
+t.create('daily downloads (expected failure)')
+  .get('/dd/djangorestframework.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'downloads',
+    value: 'no longer available'
+  }));
+
+t.create('weekly downloads (expected failure)')
+  .get('/dw/djangorestframework.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'downloads',
+    value: 'no longer available'
+  }));
+
+t.create('monthly downloads (expected failure)')
+  .get('/dm/djangorestframework.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'downloads',
+    value: 'no longer available'
+  }));
+
+t.create('daily downloads (invalid)')
+  .get('/dd/not-a-package.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'pypi',
+    value: 'invalid'
+  }));
+
+t.create('weekly downloads (invalid)')
+  .get('/dw/not-a-package.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'pypi',
+    value: 'invalid'
+  }));
+
+t.create('monthly downloads (invalid)')
+  .get('/dm/not-a-package.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'pypi',
+    value: 'invalid'
+  }));
+
+/*
+  Note:
+  Not all project on PyPi follow SemVer
+
+  Versions strings like
+  - 2.7.3.2
+  - 2.0rc1
+  - 0.1.30b10
+  are prefectly legal.
+
+  We'll run this test againt a project that follows SemVer...
+*/
+t.create('version (semver)')
+  .get('/v/requests.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'pypi',
+    value: isSemver
+  }));
+
+/*
+  ..whereas this project uses version numbers that
+  should just be validated as an arbitary string
+*/
+t.create('version (not semver)')
+  .get('/v/psycopg2.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'pypi',
+    value: Joi.string()
+  }));
+
+t.create('version (invalid)')
+  .get('/v/not-a-package.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'pypi',
+    value: 'invalid'
+  }));
+
+t.create('licence (valid)')
+  .get('/l/requests/2.18.4.json')
```

Comment:
Could you add a license badge without the version too? And same for the other badges, like format, pyversions, etc.